sql/ddl_internal.sql

```diff
@@ -119,10 +119,18 @@ BEGIN
 
     IF num_slices IS NULL THEN
         -- Open dimension
-        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
+        IF column_type NOT IN ('BIGINT', 'INTEGER', 'SMALLINT', 'DATE', 'TIMESTAMP', 'TIMESTAMPTZ') THEN
             RAISE EXCEPTION 'illegal type for column "%": %', column_name, column_type
             USING ERRCODE = 'IO102';
         END IF;
+        IF column_type = 'DATE'::regtype AND interval_length < _timescaledb_internal.interval_to_usec('1 day') THEN
```

Comment:
Why not have the interval just be defined as the integer number of days, as opposed to in usec?

I mean, we both require that is (a) be greater than a day, and (b) always be in day-long intervals.